### PR TITLE
[FIX] web_editor: fixes on colors and unbreakable

### PR DESCRIPTION
**Pull request by sebgeelen**
See odoo task : [2580158](https://www.odoo.com/web#id=2580158&action=333&active_id=1695&model=project.task&view_type=form&cids=1&menu_id=4720)

----------

#### [FIX] web_editor: ensure <font> colors are kept properly

color and background-color styles on `<font>` tags were being wrongly overridden when applying another color or background color to the element.

----------

#### [FIX] web_editor: allow deletion of unbreakable elements when empty

unbreakable elements were being considered as unremovable in some deletebackward cases.
with these changes it is now possible to remove a unbreakble element if it is empty

----------
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web_editor/static/lib/odoo-editor/src/commands/commands.js [hunk 1]
```diff
@@ -498,7 +498,7 @@ export const editorCommands = {
         // Color the selected <font>s and remove uncolored fonts.
         for (const font of new Set(fonts)) {
             colorElement(font, color, mode);
-            if (!hasColor(font, mode) && !hasColor(font, mode)) {
+            if (!hasColor(font, mode) && !font.hasAttribute('style')) {
```
Comment (Zinston) [inline addons/web_editor/static/lib/odoo-editor/src/commands/commands.js:501]: Does that property always exist?

File: addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js [hunk 2]
```diff
@@ -328,7 +328,7 @@ export class OdooEditor extends EventTarget {
         // Rollback if node.ouid changed. This ensures that nodes never change
         // unbreakable ancestors.
         node.ouid = node.ouid || getOuid(node, true);
-        if (testunbreak) {
+        if (testunbreak && !(node.nodeType === Node.TEXT_NODE && !node.length)) {
```
Comment (Zinston) [inline addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js:331]: Is it ok that this only considers really empty text nodes but not whitespace?

File: addons/web_editor/static/lib/odoo-editor/test/spec/editor.test.js [hunk 3]
```diff
@@ -93,6 +93,35 @@ describe('Editor', () => {
...
+                        stepFunction: async editor => {
+                            await deleteBackward(editor);
+                            await deleteBackward(editor);
+                            await deleteBackward(editor);
+                        },
+                        contentAfter: `<div>
+                                            <div>
+                                                <p>AB[]</p></div></div>`,
+                    });
+                });
```
Comment (Zinston) [inline addons/web_editor/static/lib/odoo-editor/test/spec/editor.test.js:124]: Nice.

**Review by Zynton — COMMENTED** [review]
Looks good to me. A couple of questions, just in case.
Also, I'm going to need you to expand your commit messages so they respect the guidelines (reference to task in each one, proper capitalization), and to retarget to saas-14.3.

File: addons/web_editor/static/lib/odoo-editor/src/commands/commands.js (hunk 1 above)
Comment (sebgeelen) [reply addons/web_editor/static/lib/odoo-editor/src/commands/commands.js:501]: `fonts` is an array of fontElement.

so Yes

File: addons/web_editor/static/lib/odoo-editor/test/spec/editor.test.js (hunk 3 above)
Comment (sebgeelen) [reply addons/web_editor/static/lib/odoo-editor/test/spec/editor.test.js:124]: :shrug:

File: addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js (hunk 2 above)
Comment (sebgeelen) [reply addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js:331]: This seems to be enough to deal with the problems emerging from formatted HTML, and it is also the least invasive check.

So IMO it is OK as long as we don't find other related problems

cc. @Goaman, as he is The One Who Found The Fix. #praise

Comment (Zinston) [reply addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js:331]: OK. But did you actually test with whitespace? If that doesn't work, wouldn't `isVisibleStr` be our go-to function for this sort of situation?

Comment (sebgeelen) [reply addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js:331]: I've added some test to ensure it work properly in all cases ( in the retargeted PR #73541 )

**Comment by sebgeelen** [comment]
Retargeted to saas14.3 in #73541

**Pull request closed without merging**